sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/BatchSpannerRead.java

```diff
@@ -251,6 +259,22 @@ public void processElement(ProcessContext c) throws Exception {
         throw (e);
       }
       serviceCallMetric.call("ok");
+      // Report Lineage metrics
+      @Nullable String tableName = op.getReadOperation().tryGetTableName();
+      if (!Objects.equals(reportedLineage, tableName)) {
+        ImmutableList.Builder<String> segments =
+            ImmutableList.<String>builder()
+                .add(
+                    projectId,
+                    spannerAccessor.getInstanceConfigId(),
```

Comment:
Per doc spanner FQN has format `spanner:{projectId}.{instanceConfigId}.{instanceId}.{databaseId}.{assetId}`

However, currently instanceConfigId is not mentioned throughout SpannerIO. Had to make a real API call to get instance config Id.

cc: @manitgupta would like to learn if there a better way to do this.